### ISSUE TITLE
Incorrect entries listed as "Unable to satisfy the following requirements" re: crystal version

### ISSUE THREAD

**Issue by Sija** [crystal-lang/shards#491]
When issuing `shards install/update` command, shards are refusing to cooperate due to unsupported crystal version in some of the dependencies and I'm greeted with a list of shards that are "Unable to satisfy the following requirements":

```
- `crystal (>= 0.35.0)` required by `raven 1.9.1`
- `crystal (>= 0.36.0)` required by `kemal 1.0.0`
- `crystal (>= 0.35.0)` required by `ameba 0.14.1`
```

**Comment by asterite** [comment]
Isn't this expected? Or what's unexpected?

**Comment by Sija** [comment]
`>= 0.35.0` restriction satisfies version number `1.0.0`, so what's unexpected is seeing shards with such restrictions listed there.

**Comment by Sija** [comment]
It obviously does. I don't follow to which comment you're referring to.

**Comment by Sija** [comment]
@asterite AFAIU it relates to the `~>` operator.

**Comment by Sija** [comment]
> This correctly identifies a number of mismatches in Crystal requirements.

@straight-shoota It doesn't, look at the requirements for `raven` and `ameba` shards.

**Comment by straight-shoota** [comment]
Indeed, you're right.

**Comment by asterite** [comment]
Maybe it's listing those shards so you can understand the conflict?

**Comment by straight-shoota** [comment]
Yeah, but `>= 0.35.1` should be subsumed by all other restrictions except `< 1.0.0`. And even that is not a conflict.

**Comment by Sija** [comment]
This bug results in spreading confusion re: the version selectors, see https://github.com/crystal-community/timecop.cr/pull/7 as another instance of that.
